### PR TITLE
fix for repeated enums; test for repeated enums

### PR DESCRIPTION
**Pull request by JohnF42**
Repeated enums are currently creating invalid code. This fixes this and adds a test.

### CODE REVIEW

**Comment by zaquestion** [comment]
@lelandbatey @adamryman Can you take a look? Are there implications around parsing the request here?

File: cmd/_integration-tests/cli/test-service-definitions/2-repeated/repeated.proto [hunk 1]
```diff
@@ -20,6 +20,13 @@ service TEST {
   }
 }
 
+enum TestType {
```
Comment (adamryman) [inline cmd/_integration-tests/cli/test-service-definitions/2-repeated/repeated.proto:23]: I would prefer `EnumType` or `EnumTestType`.

File: gengokit/httptransport/httptransport.go [hunk 2]
```diff
@@ -276,7 +276,7 @@ func createDecodeConvertFunc(f Field) (string, bool) {
 		needsErrorCheck = false
 	}
 
-	if f.IsEnum {
+	if f.IsEnum && !f.Repeated {
```
Comment (adamryman) [inline gengokit/httptransport/httptransport.go:279]: Consider adding a comment when you run across something like this that might be unexpected.

File: cmd/_integration-tests/cli/test-service-definitions/2-repeated/repeated.proto [hunk 3]
```diff
@@ -20,6 +20,13 @@ service TEST {
...
+  TEST1 = 0;
+  TEST2 = 1;
+  TEST3 = 2;
+}
+
+
```
Comment (adamryman) [inline cmd/_integration-tests/cli/test-service-definitions/2-repeated/repeated.proto:29]: Possibly unneeded whitespace.

**Review by adamryman — APPROVED** [review]
Great! :100: :1st_place_medal: :fireworks: 

@JohnF42 the comments I left are for future considerations. I believe they would improve things, though the functionality you added is just fine without them.

**Comment by adamryman** [comment]
@zaquestion this will have no implications around parsing the request. As we only support Enums in body of a request, and the json parser will handle that for us.

@JohnF42 great catch here! We were not testing repeated Enums and so we did not generate the "slice/array" code in the case the Enum was repeated.

Thanks for this addition, and especially than you for the test.

**Comment by zaquestion** [comment]
@adamryman You sure about that? https://github.com/TuneLab/truss/pull/176/files

**Comment by adamryman** [comment]
@zaquestion hmmm.

Could put the warning back in for now if someone tries to put a repeated Enum in the path or query.

Chaninging this:
https://github.com/TuneLab/truss/blob/master/gengokit/httptransport/httptransport.go#L115
```  
        if newField.IsEnum {
            continue
        }
```
to this
```
        if newField.IsEnum && !newField.Repeated {
            continue
        }
```

**Pull request merged**